freespace2/freespace.cpp

```diff
@@ -7522,8 +7522,48 @@ void game_title_screen_display()
 	
 	Game_title_logo = bm_load(Game_logo_screen_fname[gr_screen.res]);
 
-	if (!Splash_screens.empty()) {
-		Game_title_bitmap = bm_load(Splash_screens[Random::next(0, (int)Splash_screens.size() - 1)].c_str());
+	// Search splash screens for matches
+	bool any_exact_match = false;
+	bool any_range_match = false;
+	for (auto &ss : Splash_screens) {
+		if (ss.aspect_ratio_exact != 0.0f) {
+			ss.match_exact = fl_equal(ss.aspect_ratio_exact, gr_screen.clip_aspect, 0.01f);
+		}
+		if (ss.aspect_ratio_min != 0.0f || ss.aspect_ratio_max != 0.0f) {
+			bool match = true;
+			if (ss.aspect_ratio_min != 0.0f && ss.aspect_ratio_min > gr_screen.clip_aspect) {
+				match = false;
+			}
+			if (ss.aspect_ratio_max != 0.0f && ss.aspect_ratio_max < gr_screen.clip_aspect) {
+				match = false;
+			}
+			ss.match_range = match;
+		}
+		if (ss.match_exact) {
+			any_exact_match = true;
+		}
+		if (ss.match_range) {
+			any_range_match = true;
+		}
+	}
+
+	// Filter splash screens
+	Splash_screens.erase(
+		std::remove_if(Splash_screens.begin(), Splash_screens.end(), [any_exact_match, any_range_match](const splash_screen& ss) {
+			if (any_exact_match) {
+				return !ss.match_exact;
+			} else if (any_range_match) {
+				return !ss.match_range;
+			} else {
+				return !ss.is_default;
+			}
+		}), Splash_screens.end()
+	);
+
+	if (Splash_screens.size() == 1) {
+		Game_title_bitmap = bm_load(Splash_screens[0].filename.c_str());
+	} else if (!Splash_screens.empty()) {
+		Game_title_bitmap = bm_load(Splash_screens[Random::next((int)Splash_screens.size())].filename.c_str());
```

Comment:
Is it sensible to have two lists or a priority sorted list, to basically allow the game to reliably pick an exact match over a range match that may be meant as a backup?

Reply:
The section labeled `// Search splash screens for matches` already does this.  If one or more exact matches exist, it picks from them.  If there are no exact matches but there are range matches, it picks from that list.  If there are no matches of any type, it picks from the defaults.  If there are no matches and no defaults, it fallse back to `Preload` and `2_Preload`.

Reply:
Ah, I see